### PR TITLE
feat: implement Safe Contract sdk execution

### PR DESCRIPTION
**Pull request by jmdev3**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

- closes 0xsquid/squid-core# (issue number)
- Zenhub issue: (optional)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (code improvement)
- [ ] Styles (adding, editing or fixing styles)
- [ ] Unit test

# How Has This Been Tested?

Please describe the steps to test.

## Evidence

Please add some evidence like screenshots or records.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by odcey — APPROVED** [review]
Looks good to me

File: src/index.ts [hunk 1]
```diff
@@ -30,6 +33,8 @@ import { parseSdkInfoResponse } from "./0xsquid/v1/sdk-info";
 import { parseRouteResponse } from "./0xsquid/v1/route";
 import { parseStatusResponse } from "./0xsquid/v1/status";
 
+const appsSdk = new SafeAppsSDK({ debug: true });
```
Comment (alexandrewolff) [inline src/index.ts:36]: Aren't we supposed to provide 'allowedDomains: [/gnosis-safe.io$/, /app.safe.global$/],' as well ?

Comment (alexandrewolff) [reply src/index.ts:36]: Plus can't we pass this to the function from whatever we get from wagmi ? @odcey

Comment (odcey) [reply src/index.ts:36]: > Aren't we supposed to provide 'allowedDomains: [/gnosis-safe.io$/, /app.safe.global$/],' as well ?

Agree, it's better to add this as an option in the constructor

File: src/index.ts [hunk 2]
```diff
@@ -254,8 +255,11 @@ export class Squid {
     signer,
     route,
     executionSettings,
-    overrides
-  }: ExecuteRoute): Promise<ethers.providers.TransactionResponse> {
+    overrides,
+    safeContext
+  }: ExecuteRoute): Promise<
```
Comment (rajat43) [inline src/index.ts:260]: Should safeContext be optional Param here in executeRoute?

**Review by rajat43 — CHANGES_REQUESTED** [review]

**Review by rajat43 — APPROVED** [review]
Lgtm

**Pull request merged**